rc/filetype/elvish.kak

```diff
@@ -0,0 +1,57 @@
+# Syntax highlighting and indentation for Elvish (https://elv.sh)
+
+hook global BufCreate .*\.elv %{
+    set-option buffer filetype elvish
+}
+
+hook global WinSetOption filetype=elvish %<
+    require-module elvish
+
+    hook window InsertChar \n -group elvish-insert elvish-indent
+    hook window InsertChar [\]})] -group elvish-insert elvish-deindent
+    hook -once -always window WinSetOption filetype=.* %{ remove-hooks window elvish-.+ }
+>
+
+hook -group elvish-highlight global WinSetOption filetype=elvish %{
+    add-highlighter window/elvish ref elvish
+    hook -once -always window WinSetOption filetype=.* %{ remove-highlighter window/elvish }
+}
+
+provide-module elvish %§
+
+add-highlighter shared/elvish regions
+add-highlighter shared/elvish/code default-region group
+add-highlighter shared/elvish/double_string region '"' (?<!\\)(\\\\)*" fill string
+add-highlighter shared/elvish/single_string region "'" (?<!')('')*' fill string
+add-highlighter shared/elvish/comment region '#' $ fill comment
+
+add-highlighter shared/elvish/code/variable regex \$[\w\d-_:~]+ 0:variable
+add-highlighter shared/elvish/code/variable_in_assignment regex (?:^|\{\s|\(|\|)\h*(?:var|set|tmp|del)((?:\h+[\w\d-_:~]+)*) 1:variable
```

Comment:
I guess for completeness we could add `;` as statement separator

```suggestion
add-highlighter shared/elvish/code/variable_in_assignment regex (?:^|\{\s|\(|\||;)\h*(?:var|set|tmp|del)((?:\h+[\w\d-_:~]+)*) 1:variable
```

Reply:
Added; the `;` requires escaping so it should be `\;`.